### ISSUE TITLE
Question: Yolo possible for Image-Classification instead of Object-Detection?

### ISSUE THREAD

**Issue by flowzen1337** [AlexeyAB/darknet#6055]
Hi @AlexeyAB 

i came across a problem where i used tensorflow for but i want to do it all in YOLO itself without using tensorflow.
I want to classify a image for example as "image of cars" instead of detecting each object "car" in the image.

Can i do it also with yolo, just annotated a bunch of pictures like i do it with objects?
I thought about cropping the whole picture as object class "image of cars" with it's whole dimension (if image got 1280x720 i would annotate the whol 1280x720 as the "image-classification-object") like in this example:

![car_classification](https://user-images.githubusercontent.com/49192965/85528812-af971b80-b60c-11ea-92c1-9092a8b5c5c7.png)

Instead of cropping each object car (for what yolo is meanly supposed to be i guess):
![car_detection](https://user-images.githubusercontent.com/49192965/85528860-bde53780-b60c-11ea-8502-2639582fa0a2.png)

So my questions are:
a) Is it possible with yolo to classify a whole picture?
b) Is it possible like explained above this?
c) Would you recommend a other of doing it?

Thanks for your help!
Cheers

**Comment by ravi02512** [comment]
Hi flowzen1337,
Yes it is possible to perform image classification with darknet repo and there is also bunch of choices of network available.

**Comment by flowzen1337** [comment]
> Hi flowzen1337,
> Yes it is possible to perform image classification with darknet repo and there is also bunch of networks available for image classification problem.

Thanks @ravi02512 !

Can you point me in the right direction (any tipps would be helpfull!) if my idea from above is possible with the same training procedure like for object detection, just annotating the whole image instead of the objects in the image? 

Which network would you recommend for doing it?

Thanks for your help 💯 🥇 :-)

**Comment by 1027663760** [comment]
look：https://pjreddie.com/darknet/imagenet/

**Comment by lizq-git** [comment]
> look：https://pjreddie.com/darknet/imagenet/

Hi, 1027, I believed the question is how do train a classifier from via darknet, I am running into the same issue, the website that you have shared doesn't contain the idea of "transfer learning" via the provided .cfg and .weight file

**Comment by lizq-git** [comment]
Hi, Flowzen.

I am facing this issue as well.

The reason I wanted to have a classifier via darknet is that darknet C++ api is more graceful than TensorFlow.

So, I have tried out what you have described today.

I have 5 classes, I faked the "class xmin xmax ymin ymax' information, e.g. class 1 has 0, imgWidth*0.1, 0, imgHeight*0.1, then class 2 has 0, imgWidth*0.2, 0 imgWidth*0.2.

The loss is beautiful, but the result is ugly, as the bounding box is happening everywhere.

So, I went on to the official tutorial https://pjreddie.com/darknet/ - train a classifer on Cifar10, the top 1 accuracy for my dataset is ~60%.

And there are still VGG16.cfg, but when I used it, the loss became inf after 250ish iteration.

I am stilling looking for a solution for this, how about you? Have you solved your problem?
